### ISSUE TITLE
[package] boost/1.74.0: Boost.Locale failed to find iconv (Android)

### ISSUE THREAD

**Issue by granger35** [conan-io/conan-center-index#7114]
### Package and Environment Details
  * Package **boost/1.74.0**
  * Operating System+version: **Linux Ubuntu 18.04**
  * Compiler+version: **clang 8**
  * Docker image: **conanio/android-clang_8-armv8**
  * Conan version: **conan 1.39.0**
  
### Conan profile 
* build:
```
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++
compiler.version=7
os=Linux
compiler.cppstd=17
[options]
[conf]
[build_requires]
[env]

```
* host
```
[settings]
arch=armv8
build_type=Release
compiler=clang
compiler.libcxx=libc++
compiler.version=8
os=Android
os.api_level=21
compiler.cppstd=17
[options]
[conf]
[build_requires]
[env]
```

### Steps to reproduce
conan install boost/1.74.0@ --build=missing --update -pr:b=build -pr:h=host

### Logs 
<details><summary>Click to expand log</summary>

```
conan@d52884b71ed7:~$ conan install boost/1.74.0@ --build=missing --update -pr:b=build -pr:h=host
Configuration (profile_host):
[settings]
arch=armv8
build_type=Release
compiler=clang
compiler.cppstd=17
compiler.libcxx=libc++
compiler.version=8
os=Android
os.api_level=21
[options]
[build_requires]
[env]

Configuration (profile_build):
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=17
compiler.libcxx=libstdc++
compiler.version=7
os=Linux
[options]
[build_requires]
[env]

bzip2/1.0.8: WARN: Can't update, no package in remote
zlib/1.2.11: WARN: Can't update, no package in remote
boost/1.74.0: WARN: Package binary is corrupted, removing: e45c5c16747667b89274c3279a1f47ccf4daa1f9
Installing package: boost/1.74.0
Requirements
    boost/1.74.0 from 'conan-bcom' - Cache
    bzip2/1.0.8 from 'conancenter' - Cache
    libbacktrace/cci.20210118 from 'conan-bcom' - Cache
    zlib/1.2.11 from 'conan-bcom' - Cache
Packages
    boost/1.74.0:e45c5c16747667b89274c3279a1f47ccf4daa1f9 - Build
    bzip2/1.0.8:d0202e15ed2bffe00fb2ab7504e436269af6ec9f - Cache
    libbacktrace/cci.20210118:3cc1debdebdc6ad0796bb6699c1ec121d98add89 - Cache
    zlib/1.2.11:3cc1debdebdc6ad0796bb6699c1ec121d98add89 - Cache
Build requirements
    b2/4.5.0 from 'conan-solar' - Cache
Build requirements packages
    b2/4.5.0:4db1be536558d833e52e862fd84d64d75c2b3656 - Cache

Cross-build from 'Linux:x86_64' to 'Android:armv8'
Installing (downloading, building) binaries...
b2/4.5.0: Already installed!
bzip2/1.0.8: Already installed!
bzip2/1.0.8: Appending PATH environment variable: /home/conan/.conan/data/bzip2/1.0.8/_/_/package/d0202e15ed2bffe00fb2ab7504e436269af6ec9f/bin
libbacktrace/cci.20210118: Already installed!
zlib/1.2.11: Already installed!
boost/1.74.0: Applying build-requirement: b2/4.5.0
boost/1.74.0: WARN: Build folder is dirty, removing it: /home/conan/.conan/data/boost/1.74.0/_/_/build/e45c5c16747667b89274c3279a1f47ccf4daa1f9
boost/1.74.0: Building your package in /home/conan/.conan/data/boost/1.74.0/_/_/build/e45c5c16747667b89274c3279a1f47ccf4daa1f9
boost/1.74.0: Generator txt created conanbuildinfo.txt
boost/1.74.0: Aggregating env generators
boost/1.74.0: Calling build()
WARN: replace_in_file didn't find pattern '$(>) > $(<)' in '/home/conan/.conan/data/boost/1.74.0/_/_/source/source_subfolder/libs/stacktrace/build/Jamfile.v2' file.
WARN: replace_in_file didn't find pattern '/* thread_local */' in '/home/conan/.conan/data/boost/1.74.0/_/_/source/source_subfolder/boost/stacktrace/detail/libbacktrace_impls.hpp' file.
WARN: replace_in_file didn't find pattern '/* static __thread */' in '/home/conan/.conan/data/boost/1.74.0/_/_/source/source_subfolder/boost/stacktrace/detail/libbacktrace_impls.hpp' file.
boost/1.74.0: WARN: Patching user-config.jam
boost/1.74.0: WARN: 
using zlib : 1.2.11 : <include>"/home/conan/.conan/data/zlib/1.2.11/_/_/package/3cc1debdebdc6ad0796bb6699c1ec121d98add89/include" <search>"/home/conan/.conan/data/zlib/1.2.11/_/_/package/3cc1debdebdc6ad0796bb6699c1ec121d98add89/lib" <name>z ;
using bzip2 : 1.0.8 : <include>"/home/conan/.conan/data/bzip2/1.0.8/_/_/package/d0202e15ed2bffe00fb2ab7504e436269af6ec9f/include" <search>"/home/conan/.conan/data/bzip2/1.0.8/_/_/package/d0202e15ed2bffe00fb2ab7504e436269af6ec9f/lib" <name>bz2 ;
using "clang-linux" :  :  /android-ndk-r19c/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android21-clang++ : 
<archiver>"/android-ndk-r19c/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android-ar" <ranlib>"/android-ndk-r19c/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android-ranlib" <compileflags>"-I/home/conan/.conan/data/libbacktrace/cci.20210118/_/_/package/3cc1debdebdc6ad0796bb6699c1ec121d98add89/include" <linkflags>"-L/home/conan/.conan/data/libbacktrace/cci.20210118/_/_/package/3cc1debdebdc6ad0796bb6699c1ec121d98add89/lib"  ;
boost/1.74.0: Cross building, detecting compiler...
boost/1.74.0: Cross building flags: []
boost/1.74.0: WARN: b2 -q numa=on target-os=android architecture=arm address-model=64 binary-format=elf abi=aapcs --layout=system --user-config=/home/conan/.conan/data/boost/1.74.0/_/_/source/source_subfolder/tools/build/user-config.jam -sNO_ZLIB=0 -sNO_BZIP2=0 -sNO_LZMA=1 -sNO_ZSTD=1 boost.locale.icu=off --disable-icu boost.locale.iconv=on boost.locale.iconv.lib=libc threading=multi visibility=hidden link=static variant=release --with-atomic --with-chrono --with-container --with-context --with-contract --with-coroutine --with-date_time --with-exception --with-fiber --with-filesystem --with-graph --with-iostreams --with-locale --with-log --with-math --with-nowide --with-program_options --with-random --with-regex --with-serialization --with-stacktrace --with-system --with-test --with-thread --with-timer --with-type_erasure --with-wave toolset=clang-linux cxxflags=-std=c++17 pch=on  cxxflags="-fPIC -DBOOST_STACKTRACE_ADDR2LINE_LOCATION=/usr/bin/addr2line" install --prefix=/home/conan/.conan/data/boost/1.74.0/_/_/package/e45c5c16747667b89274c3279a1f47ccf4daa1f9 -j6 --abbreviate-paths -d0 --debug-configuration --build-dir="/home/conan/.conan/data/boost/1.74.0/_/_/build/e45c5c16747667b89274c3279a1f47ccf4daa1f9"
notice: found boost-build.jam at /home/conan/.conan/data/boost/1.74.0/_/_/source/source_subfolder/boost-build.jam
notice: loading B2 from /home/conan/.conan/data/boost/1.74.0/_/_/source/source_subfolder/tools/build/src/bootstrap.jam
warning: mismatched versions of B2 engine and core
warning: B2 engine (b2) is 4.5.0
warning: B2 core (at /home/conan/.conan/data/boost/1.74.0/_/_/source/source_subfolder/tools/build/src) is 4.3-git
notice: Searching '/etc' '/home/conan' '/home/conan/.conan/data/boost/1.74.0/_/_/source/source_subfolder/tools/build/src' '/home/conan/.conan/data/boost/1.74.0/_/_/source/source_subfolder/tools/build/src/kernel' '/home/conan/.conan/data/boost/1.74.0/_/_/source/source_subfolder/tools/build/src/util' '/home/conan/.conan/data/boost/1.74.0/_/_/source/source_subfolder/tools/build/src/build' '/home/conan/.conan/data/boost/1.74.0/_/_/source/source_subfolder/tools/build/src/tools' '/home/conan/.conan/data/boost/1.74.0/_/_/source/source_subfolder/tools/build/src/contrib' '/home/conan/.conan/data/boost/1.74.0/_/_/source/source_subfolder/tools/build/src/.' for site-config configuration file 'site-config.jam'.
notice: Configuration file 'site-config.jam' not found in '/etc' '/home/conan' '/home/conan/.conan/data/boost/1.74.0/_/_/source/source_subfolder/tools/build/src' '/home/conan/.conan/data/boost/1.74.0/_/_/source/source_subfolder/tools/build/src/kernel' '/home/conan/.conan/data/boost/1.74.0/_/_/source/source_subfolder/tools/build/src/util' '/home/conan/.conan/data/boost/1.74.0/_/_/source/source_subfolder/tools/build/src/build' '/home/conan/.conan/data/boost/1.74.0/_/_/source/source_subfolder/tools/build/src/tools' '/home/conan/.conan/data/boost/1.74.0/_/_/source/source_subfolder/tools/build/src/contrib' '/home/conan/.conan/data/boost/1.74.0/_/_/source/source_subfolder/tools/build/src/.'.
notice: Loading explicitly specified user configuration file:
    /home/conan/.conan/data/boost/1.74.0/_/_/source/source_subfolder/tools/build/user-config.jam
notice: Searching '/home/conan/.conan/data/boost/1.74.0/_/_/source/source_subfolder/tools/build' for user-config configuration file 'user-config.jam'.
notice: Loading user-config configuration file 'user-config.jam' from '/home/conan/.conan/data/boost/1.74.0/_/_/source/source_subfolder/tools/build'.
notice: [zlib] Using pre-installed library
notice: [zlib] Condition
notice: [bzip2] Using pre-installed library
notice: [bzip2] Condition
notice: will use '/android-ndk-r19c/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android21-clang++' for clang-linux, condition <toolset>clang-linux-8.0.2
notice: [zlib] zlib is already configured
notice: [bzip2] bzip is already configured
notice: iostreams: not using lzma compression 
notice: iostreams: not using zstd compression 
notice: [python-cfg] Configuring python...
notice: [python-cfg] Checking interpreter command "python"...
notice: [python-cfg] running command 'python -c "from sys import *; print('version=%d.%d\nplatform=%s\nbase_prefix=%s\nbase_exec_prefix=%s\nexecutable=%s' % (version_info[0],version_info[1],platform,base_prefix,base_exec_prefix,executable))" 2>&1'
notice: [python-cfg] ...requested configuration matched!
notice: [python-cfg] Details of this Python configuration:
notice: [python-cfg]   interpreter command: "python"
notice: [python-cfg]   include path: "/opt/pyenv/versions/3.7.5/include/python3.7"
notice: [python-cfg]   library path: "/opt/pyenv/versions/3.7.5/lib/python3.7/config" "/opt/pyenv/versions/3.7.5/lib"
notice: [python-cfg] Checking for NumPy...
notice: [python-cfg] running command 'python -c "import sys; sys.stderr = sys.stdout; import numpy; print(numpy.get_include())"'
notice: [python-cfg] NumPy disabled. Reason:
notice: [python-cfg]   python -c "import sys; sys.stderr = sys.stdout; import numpy; print(numpy.get_include())" aborted with 
notice: [python-cfg]   Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'numpy'
Performing configuration checks

    - default address-model    : 64-bit
    - default architecture     : arm
    - C++11 mutex              : yes
    - lockfree boost::atomic_flag : yes
    - has stat::st_mtim        : yes
    - has stat::st_mtimensec   : yes
    - has stat::st_mtimespec   : no
    - Boost.Config Feature Check: cxx11_auto_declarations : yes
    - Boost.Config Feature Check: cxx11_constexpr : yes
    - Boost.Config Feature Check: cxx11_defaulted_functions : yes
    - Boost.Config Feature Check: cxx11_final : yes
    - Boost.Config Feature Check: cxx11_hdr_mutex : yes
    - Boost.Config Feature Check: cxx11_hdr_tuple : yes
    - Boost.Config Feature Check: cxx11_lambdas : yes
    - Boost.Config Feature Check: cxx11_noexcept : yes
    - Boost.Config Feature Check: cxx11_nullptr : yes
    - Boost.Config Feature Check: cxx11_rvalue_references : yes
    - Boost.Config Feature Check: cxx11_template_aliases : yes
    - Boost.Config Feature Check: cxx11_thread_local : yes
    - Boost.Config Feature Check: cxx11_variadic_templates : yes
    - has_icu builds           : no
    - zlib                     : yes
    - bzip2                    : yes
    - iconv (libc)             : no
- Boost.Locale failed to find iconv (libc) library to be built.
boost/1.74.0: 
boost/1.74.0: ERROR: Package 'e45c5c16747667b89274c3279a1f47ccf4daa1f9' build failed
boost/1.74.0: WARN: Build folder /home/conan/.conan/data/boost/1.74.0/_/_/build/e45c5c16747667b89274c3279a1f47ccf4daa1f9
ERROR: boost/1.74.0: Error in build() method, line 807
	self.run(full_command, run_environment=True)
	ConanException: Error 1 while executing b2 -q numa=on target-os=android architecture=arm address-model=64 binary-format=elf abi=aapcs --layout=system --user-config=/home/conan/.conan/data/boost/1.74.0/_/_/source/source_subfolder/tools/build/user-config.jam -sNO_ZLIB=0 -sNO_BZIP2=0 -sNO_LZMA=1 -sNO_ZSTD=1 boost.locale.icu=off --disable-icu boost.locale.iconv=on boost.locale.iconv.lib=libc threading=multi visibility=hidden link=static variant=release --with-atomic --with-chrono --with-container --with-context --with-contract --with-coroutine --with-date_time --with-exception --with-fiber --with-filesystem --with-graph --with-iostreams --with-locale --with-log --with-math --with-nowide --with-program_options --with-random --with-regex --with-serialization --with-stacktrace --with-system --with-test --with-thread --with-timer --with-type_erasure --with-wave toolset=clang-linux cxxflags=-std=c++17 pch=on  cxxflags="-fPIC -DBOOST_STACKTRACE_ADDR2LINE_LOCATION=/usr/bin/addr2line" install --prefix=/home/conan/.conan/data/boost/1.74.0/_/_/package/e45c5c16747667b89274c3279a1f47ccf4daa1f9 -j6 --abbreviate-paths -d0 --debug-configuration --build-dir="/home/conan/.conan/data/boost/1.74.0/_/_/build/e45c5c16747667b89274c3279a1f47ccf4daa1f9"
```

</details>

**Comment by dmn-star** [comment]
I have always turned off the three below, maybe it's an option for you too.

boost:without_locale = True
boost:without_log = True
boost:without_stacktrace = True

**Comment by granger35** [comment]
Thank you @dmn-star ,

You are right, without_locale and without_log turned to True do the job, but i need the boost logs (and logs are required if without_local is set to True).
I have to set i18n_backend_iconv to libiconv for now, default libc does not work.

**Comment by SSE4** [comment]
checking the config.log I see the error:
```
  6 Undefined symbols for architecture arm64:
  7   "_iconv_open", referenced from:
  8       _main in has_iconv_libc_obj.o
  9 ld: symbol(s) not found for architecture arm64
```
perhaps, android's libc (bionic) doesn't provide iconv interface? let's check, and if so, for Android we should use libiconv as a default backend (instead of libc iconv).

**Comment by SSE4** [comment]
found answer: https://github.com/android/ndk/issues/702
TLDR: iconv is provided by bionic since API level 28
for lower API levels we should default to `i18n_backend_iconv = "libiconv"`
I'll prepare the fix

**Comment by granger35** [comment]
Thank you for pointing out the problem and fixing it.

**Comment by JohnVida** [comment]
How to fix this issue in Linux?
### ### 



======== Installing packages ========
b2/4.9.6: Already installed! (1 of 6)
bzip2/1.0.8: Already installed! (2 of 6)
fmt/8.0.1: Already installed! (3 of 6)
gsl-lite/0.37.0: Already installed! (4 of 6)
zlib/1.2.13: Already installed! (5 of 6)

-------- Installing package boost/1.71.0 (6 of 6) --------
boost/1.71.0: Building from source
boost/1.71.0: Package boost/1.71.0:fbf39d5786e603f013aa4437f27a29522ee048c9
boost/1.71.0: Building your package in /home/jiqiang/.conan2/p/t/boost6d4ff9145f3c4/b
boost/1.71.0: Calling generate()
boost/1.71.0: Generators folder: /home/jiqiang/.conan2/p/t/boost6d4ff9145f3c4/b/build-debug/conan
boost/1.71.0: Generating aggregated env files
boost/1.71.0: Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']
boost/1.71.0: Calling build()
boost/1.71.0: WARN: replace_in_file didn't find pattern '/* thread_local */' in '/home/jiqiang/.conan2/p/boost4d7c500cb2cf1/s/src/boost/stacktrace/detail/libbacktrace_impls.hpp' file.
boost/1.71.0: WARN: replace_in_file didn't find pattern '/* static __thread */' in '/home/jiqiang/.conan2/p/boost4d7c500cb2cf1/s/src/boost/stacktrace/detail/libbacktrace_impls.hpp' file.
boost/1.71.0: WARN: replace_in_file didn't find pattern 'local generic-os = [ set.difference $(all-os) : aix darwin vxworks solaris osf hpux ] ;' in '/home/jiqiang/.conan2/p/boost4d7c500cb2cf1/s/src/tools/build/src/tools/gcc.jam' file.
boost/1.71.0: WARN: replace_in_file didn't find pattern 'local no-threading = android beos haiku sgi darwin vxworks ;' in '/home/jiqiang/.conan2/p/boost4d7c500cb2cf1/s/src/tools/build/src/tools/gcc.jam' file.
boost/1.71.0: WARN: replace_in_file didn't find pattern '    <conditional>@numa' in '/home/jiqiang/.conan2/p/boost4d7c500cb2cf1/s/src/libs/fiber/build/Jamfile.v2' file.
boost/1.71.0: WARN: Patching user-config.jam
boost/1.71.0: WARN: 
using zlib : 1.2.13 : <include>"/home/jiqiang/.conan2/p/zlibdd8945e81d91f/p/include" <search>"/home/jiqiang/.conan2/p/zlibdd8945e81d91f/p/lib" <name>z ;
using bzip2 : 1.0.8 : <include>"/home/jiqiang/.conan2/p/bzip2e676af0de9f57/p/include" <search>"/home/jiqiang/.conan2/p/bzip2e676af0de9f57/p/lib" <name>bz2 ;
using "gcc" :  :  "/usr/bin/g++" : 
 ;
boost/1.71.0: Cross building, detecting compiler...
boost/1.71.0: Cross building flags: []
boost/1.71.0: WARN: b2 -q numa=on target-os=linux architecture=x86 address-model=32 binary-format=elf abi=sysv --layout=system --user-config=/home/jiqiang/.conan2/p/boost4d7c500cb2cf1/s/src/tools/build/user-config.jam -sNO_ZLIB=0 -sNO_BZIP2=0 -sNO_LZMA=1 -sNO_ZSTD=1 boost.locale.icu=off --disable-icu boost.locale.iconv=on boost.locale.iconv.lib=libc threading=multi visibility=hidden link=static variant=debug --with-atomic --with-chrono --with-container --with-context --with-contract --with-coroutine --with-date_time --with-exception --with-fiber --with-filesystem --with-graph --with-iostreams --with-locale --with-log --with-math --with-program_options --with-random --with-regex --with-serialization --with-system --with-test --with-thread --with-timer --with-type_erasure --with-wave toolset=gcc define=_GLIBCXX_USE_CXX11_ABI=1 pch=on linkflags="" cxxflags="-fPIC" install --prefix=/home/jiqiang/.conan2/p/t/boost6d4ff9145f3c4/p -j2 --abbreviate-paths -d0 --debug-configuration --build-dir="/home/jiqiang/.conan2/p/t/boost6d4ff9145f3c4/b/build-debug"
boost/1.71.0: RUN: b2 -q numa=on target-os=linux architecture=x86 address-model=32 binary-format=elf abi=sysv --layout=system --user-config=/home/jiqiang/.conan2/p/boost4d7c500cb2cf1/s/src/tools/build/user-config.jam -sNO_ZLIB=0 -sNO_BZIP2=0 -sNO_LZMA=1 -sNO_ZSTD=1 boost.locale.icu=off --disable-icu boost.locale.iconv=on boost.locale.iconv.lib=libc threading=multi visibility=hidden link=static variant=debug --with-atomic --with-chrono --with-container --with-context --with-contract --with-coroutine --with-date_time --with-exception --with-fiber --with-filesystem --with-graph --with-iostreams --with-locale --with-log --with-math --with-program_options --with-random --with-regex --with-serialization --with-system --with-test --with-thread --with-timer --with-type_erasure --with-wave toolset=gcc define=_GLIBCXX_USE_CXX11_ABI=1 pch=on linkflags="" cxxflags="-fPIC" install --prefix=/home/jiqiang/.conan2/p/t/boost6d4ff9145f3c4/p -j2 --abbreviate-paths -d0 --debug-configuration --build-dir="/home/jiqiang/.conan2/p/t/boost6d4ff9145f3c4/b/build-debug"
notice: found boost-build.jam at /home/jiqiang/.conan2/p/boost4d7c500cb2cf1/s/src/boost-build.jam
notice: loading B2 from /home/jiqiang/.conan2/p/b21a8cf2a364eb1/p/bin/.b2/kernel/bootstrap.jam
notice: Searching '/etc' '/home/jiqiang' '/home/jiqiang/.conan2/p/b21a8cf2a364eb1/p/bin/.b2/kernel' '/home/jiqiang/.conan2/p/b21a8cf2a364eb1/p/bin/.b2/util' '/home/jiqiang/.conan2/p/b21a8cf2a364eb1/p/bin/.b2/build' '/home/jiqiang/.conan2/p/b21a8cf2a364eb1/p/bin/.b2/tools' '/home/jiqiang/.conan2/p/b21a8cf2a364eb1/p/bin/.b2/contrib' '/home/jiqiang/.conan2/p/b21a8cf2a364eb1/p/bin/.b2/.' for site-config configuration file 'site-config.jam'.
notice: Configuration file 'site-config.jam' not found in '/etc' '/home/jiqiang' '/home/jiqiang/.conan2/p/b21a8cf2a364eb1/p/bin/.b2/kernel' '/home/jiqiang/.conan2/p/b21a8cf2a364eb1/p/bin/.b2/util' '/home/jiqiang/.conan2/p/b21a8cf2a364eb1/p/bin/.b2/build' '/home/jiqiang/.conan2/p/b21a8cf2a364eb1/p/bin/.b2/tools' '/home/jiqiang/.conan2/p/b21a8cf2a364eb1/p/bin/.b2/contrib' '/home/jiqiang/.conan2/p/b21a8cf2a364eb1/p/bin/.b2/.'.
notice: Loading explicitly specified user configuration file:
    /home/jiqiang/.conan2/p/boost4d7c500cb2cf1/s/src/tools/build/user-config.jam
notice: Searching '/home/jiqiang/.conan2/p/boost4d7c500cb2cf1/s/src/tools/build' for user-config configuration file 'user-config.jam'.
notice: Loading user-config configuration file 'user-config.jam' from '/home/jiqiang/.conan2/p/boost4d7c500cb2cf1/s/src/tools/build'.
notice: [zlib] Using pre-installed library
notice: [zlib] Condition
notice: [bzip2] Using pre-installed library
notice: [bzip2] Condition
notice: will use '/usr/bin/g++' for gcc, condition <toolset>gcc-11
notice: using gcc libraries :: <toolset>gcc-11 :: /usr/bin /usr/lib /usr/lib32 /usr/lib64
notice: using gcc archiver :: <toolset>gcc-11 :: /usr/bin/ar
warning: toolset gcc initialization: can not find tool windres
warning: initialized from /home/jiqiang/.conan2/p/boost4d7c500cb2cf1/s/src/tools/build/user-config.jam:5
notice: using rc compiler :: <toolset>gcc-11 :: /usr/bin/as
notice: [zlib] zlib is already configured
notice: [bzip2] bzip is already configured
notice: iostreams: not using lzma compression 
notice: iostreams: not using zstd compression 
notice: [python-cfg] Configuring python...
notice: [python-cfg] Checking interpreter command "python"...
notice: [python-cfg] running command 'python -c "from sys import *; print('version=%d.%d\nplatform=%s\nprefix=%s\nexec_prefix=%s\nexecutable=%s' % (version_info[0],version_info[1],platform,prefix,exec_prefix,executable))" 2>&1'
notice: [python-cfg] ...does not invoke a working interpreter
notice: [python-cfg] Python headers and libraries not found.
Performing configuration checks

    - default address-model    : 64-bit [1]
    - default architecture     : x86 [1]
    - C++11 mutex              : no [2]
    - lockfree boost::atomic_flag : no [2]
    - Boost.Config Feature Check: cxx11_auto_declarations : no [2]
    - Boost.Config Feature Check: cxx11_constexpr : no [2]
    - Boost.Config Feature Check: cxx11_defaulted_functions : no [2]
    - Boost.Config Feature Check: cxx11_final : no [2]
    - Boost.Config Feature Check: cxx11_hdr_mutex : no [2]
    - Boost.Config Feature Check: cxx11_hdr_tuple : no [2]
    - Boost.Config Feature Check: cxx11_lambdas : no [2]
    - Boost.Config Feature Check: cxx11_noexcept : no [2]
    - Boost.Config Feature Check: cxx11_nullptr : no [2]
    - Boost.Config Feature Check: cxx11_rvalue_references : no [2]
    - Boost.Config Feature Check: cxx11_template_aliases : no [2]
    - Boost.Config Feature Check: cxx11_thread_local : no [2]
    - Boost.Config Feature Check: cxx11_variadic_templates : no [2]
    - Boost.Config Feature Check: cxx11_auto_declarations : no [3]
    - Boost.Config Feature Check: cxx11_constexpr : no [3]
    - Boost.Config Feature Check: cxx11_defaulted_functions : no [3]
    - Boost.Config Feature Check: cxx11_final : no [3]
    - Boost.Config Feature Check: cxx11_hdr_mutex : no [3]
    - Boost.Config Feature Check: cxx11_hdr_tuple : no [3]
    - Boost.Config Feature Check: cxx11_lambdas : no [3]
    - Boost.Config Feature Check: cxx11_noexcept : no [3]
    - Boost.Config Feature Check: cxx11_nullptr : no [3]
    - Boost.Config Feature Check: cxx11_rvalue_references : no [3]
    - Boost.Config Feature Check: cxx11_template_aliases : no [3]
    - Boost.Config Feature Check: cxx11_thread_local : no [3]
    - Boost.Config Feature Check: cxx11_variadic_templates : no [3]
    - has_icu builds           : no [2]
    - zlib                     : no [4]
    - bzip2                    : no [4]
    - iconv (libc)             : no [2]
- Boost.Locale failed to find iconv (libc) library to be built.

boost/1.71.0: ERROR: 
Package 'fbf39d5786e603f013aa4437f27a29522ee048c9' build failed
boost/1.71.0: WARN: Build folder /home/jiqiang/.conan2/p/t/boost6d4ff9145f3c4/b/build-debug
*********************************************************
Recipe 'boost/1.71.0' cannot build its binary
It is possible that this recipe is not Conan 2.0 ready
If the recipe comes from ConanCenter check: https://conan.io/cci-v2.html
If it is your recipe, check if it is updated to 2.0
*********************************************************

ERROR: boost/1.71.0: Error in build() method, line 887
	self.run(full_command)
	ConanException: Error 1 while executing
